identity-service/src/routes/relay.js

```diff
@@ -1,27 +1,47 @@
 const crypto = require('crypto')
 
-const { handleResponse, successResponse, errorResponseBadRequest, errorResponseServerError } = require('../apiHelpers')
+const { handleResponse, successResponse, errorResponseBadRequest, errorResponseForbidden, errorResponseServerError } = require('../apiHelpers')
 const txRelay = require('../relay/txRelay')
 const captchaMiddleware = require('../captchaMiddleware')
 const { detectAbuse } = require('../utils/antiAbuse')
 const { getFeatureFlag, FEATURE_FLAGS } = require('../featureFlag')
 
+const blockRelayAbuseErrorCodes = new Set(['3', '8', '9', '10', '11'])
+
 module.exports = function (app) {
   // TODO(roneilr): authenticate that user controls senderAddress somehow, potentially validate that
   // method sig has come from sender address as well
   app.post('/relay', captchaMiddleware, handleResponse(async (req, res, next) => {
     const body = req.body
     const redis = req.app.get('redis')
 
+    // TODO: Use auth middleware to derive this
+    const user = await models.User.findOne({
+      where: { req.senderAddress },
+      attributes: ['id', 'blockchainUserId', 'walletAddress', 'handle', 'isAbusive', 'isAbusiveErrorCode']
+    })
+
     let optimizelyClient
     let detectAbuseOnRelay = false
+    let blockAbuseOnRelay = false
     try {
       optimizelyClient = req.app.get('optimizelyClient')
       detectAbuseOnRelay = getFeatureFlag(optimizelyClient, FEATURE_FLAGS.DETECT_ABUSE_ON_RELAY)
+      blockAbuseOnRelay = getFeatureFlag(optimizelyClient, FEATURE_FLAGS.BLOCK_ABUSE_ON_RELAY)
     } catch (error) {
       req.logger.error(`failed to retrieve optimizely feature flag for detectAbuseOnRelay: ${error}`)
     }
 
+    if (
+      blockAbuseOnRelay &&
+      user?.isAbusiveErrorCode &&
```

Comment:
nit: maybe should explicitly check this against undefined in case we have 0 as an error code eventually

Reply:
we DO have 0 as a code already as a matter of fact, so this is a great callout. handling

Reply:
although that code is not being user in the block relay code set here, maybe it should be?

Reply:
adding 0 as part of the set